File: app/src/main/java/app/vodio/com/vodio/utils/MyAsyncTask.java
package app.vodio.com.vodio.utils;

import android.os.AsyncTask;

public abstract class MyAsyncTask<T>  extends AsyncTask{
    private boolean isSuccess = false;
    private OnCompleteAsyncTask onComplete;
    public MyAsyncTask(OnCompleteAsyncTask onComplete){
        this.onComplete = onComplete;
    }
    @Override
    protected abstract Object doInBackground(Object[] objects);

    protected void setSuccess(boolean s){
        isSuccess = s;
    }
    @Override
    protected void onPostExecute(Object o) {
        super.onPostExecute(o);
        if(onComplete != null) {
            if (isSuccess) {
                onComplete.onSuccess(getObject());
            } else {
                onComplete.onFail();
            }
        }
    }

    public abstract T getObject();
}

File: app/src/main/java/app/vodio/com/vodio/utils/MyAsyncTask.kt
package app.vodio.com.vodio.utils;

import android.content.Context;
import android.os.AsyncTask;

abstract class MyAsyncTask<Any>(private val onComplete : OnCompleteAsyncTask) : AsyncTask<Object, Object, Object>(){
   // private var onComplete : OnCompleteAsyncTask? = null;
    private var isSuccess : Boolean = false

    override abstract fun doInBackground(vararg p0: Object?): Object

    fun setSuccess(b : Boolean){isSuccess = b}

    override fun onPostExecute(result: Object?) {
        super.onPostExecute(result)
        if(onComplete != null) {
            val obj: Object = getObject() as Object
            when (isSuccess) {
                true -> onComplete?.onSuccess(obj)
                false -> onComplete?.onFail()
            }
        }
    }

    abstract  fun getObject(): Any
}